lib/ecto/association.ex

```diff
@@ -295,9 +295,10 @@ defmodule Ecto.Association.Has do
 
   @doc false
   def build(%{assoc: assoc, owner_key: owner_key, assoc_key: assoc_key,
-              queryable: queryable}, struct) do
+              queryable: queryable}, struct, attributes \\ %{}) do
     assoc
     |> apply(:__struct__, [])
+    |> Map.merge(Enum.into(attributes, %{}))
```

Comment:
`struct/2` does exactly this! This:

     assoc
     |> apply(:__struct__, [])
     |> Map.merge(Enum.into(attributes, %{}))

Is equivalent to:

     assoc
     |> struct(attributes)

:)